src/cargo/core/compiler/job_queue.rs

```diff
@@ -221,9 +224,19 @@ impl<'a> JobQueue<'a> {
             // jobserver interface is architected we may acquire a token that we
             // don't actually use, and if this happens just relinquish it back
             // to the jobserver itself.
-            tokens.truncate(self.active - 1);
-
-            match self.rx.recv().unwrap() {
+            tokens.truncate(self.active.len() - 1);
+
+            let count = queue_len - self.queue.len();
+            let mut active_names = self.active.iter().map(|key| match key.mode {
+                CompileMode::Doc { .. } => format!("{}(doc)", key.pkg.name()),
+                _ => key.pkg.name().to_string(),
+            }).collect::<Vec<_>>();
+            active_names.sort_unstable();
```

Comment:
I think that this may actually be easier to read as a Vec perhaps? That way names don't jump around but they should always be in the same place relative to other names. Should this also perhaps use a HashSet to deduplicate when we're building multiple targets for one crate?